### PR TITLE
Fix bug: Terminal logger Copy task retries

### LINKED ISSUE #9658: [Bug]: Copy Task retries aren't useful in Terminal Logger
### Issue Description

When the Copy Task logs warnings those warnings are held back by Terminal Logger until the entire build of that relevant project finishes. This means a viewer has no opportunity to see those messages and quickly kill the offending process, making them slightly less useful, and mostly just noisy in the final output.

### Steps to Reproduce

It's really easy to reproduce in the SDK:

* clone the SDK
* run build.cmd
* run artifacts/sdk-build-env.ps1
* launch VS Code
* run `dotnet build` on the src/Layout/redist project

### Expected Behavior

The copy errors should be visible to the user in some form earlier than the build finishing.

### Actual Behavior

The errors are buffered and emitted as a single unit:

```terminal
$ dotnet build

MSBuild version 17.10.0-preview-24060-03+1725b247e for .NET
Restore complete (1.6s)
  toolset-tasks net472 succeeded (1.0s) → E:\Code\dotnet-sdk\artifacts\tasks\bin\toolset-tasks\Debug\net472\toolset-tasks.dll
  toolset-tasks net8.0 failed with errors (13.2s)
    C:\Program Files\dotnet\sdk\9.0.100-alpha.1.24067.4\Microsoft.Common.CurrentVersion.targets(4787,5): warning MSB3026: Could not copy "E:\Code\dotnet-sdk\artifacts\tasks\obj\toolset-tasks\Debug\net8.0\toolset-tasks.dll" to "E:\Code\dotnet-sdk\artifacts\tasks\bin\toolset-tasks\Debug\net8.0\toolset-tasks.dll". Beginning retry 1 in 1000ms. The process cannot access the file 'E:\Code\dotnet-sdk\artifacts\tasks\bin\toolset-tasks\Debug\net8.0\toolset-tasks.dll' because it is being used by another process. The file is locked by: ".NET Host (19284)" [E:\Code\dotnet-sdk\src\Layout\toolset-tasks\toolset-tasks.csproj]
    C:\Program Files\dotnet\sdk\9.0.100-alpha.1.24067.4\Microsoft.Common.CurrentVersion.targets(4787,5): warning MSB3026: Could not copy "E:\Code\dotnet-sdk\artifacts\tasks\obj\toolset-tasks\Debug\net8.0\toolset-tasks.dll" to "E:\Code\dotnet-sdk\artifacts\tasks\bin\toolset-tasks\Debug\net8.0\toolset-tasks.dll". Beginning retry 2 in 1000ms. The process cannot access the file 'E:\Code\dotnet-sdk\artifacts\tasks\bin\toolset-tasks\Debug\net8.0\toolset-tasks.dll' because it is being used by another process. The file is locked by: ".NET Host (19284)" [E:\Code\dotnet-sdk\src\Layout\toolset-tasks\toolset-tasks.csproj]
    C:\Program Files\dotnet\sdk\9.0.100-alpha.1.24067.4\Microsoft.Common.CurrentVersion.targets(4787,5): warning MSB3026: Could not copy "E:\Code\dotnet-sdk\artifacts\tasks\obj\toolset-tasks\Debug\net8.0\toolset-tasks.dll" to "E:\Code\dotnet-sdk\artifacts\tasks\bin\toolset-tasks\Debug\net8.0\toolset-tasks.dll". Beginning retry 3 in 1000ms. The process cannot access the file 'E:\Code\dotnet-sdk\artifacts\tasks\bin\toolset-tasks\Debug\net8.0\toolset-tasks.dll' because it is being used by another process. The file is locked by: ".NET Host (19284)" [E:\Code\dotnet-sdk\src\Layout\toolset-tasks\toolset-tasks.csproj]
    C:\Program Files\dotnet\sdk\9.0.100-alpha.1.24067.4\Microsoft.Common.CurrentVersion.targets(4787,5): warning MSB3026: Could not copy "E:\Code\dotnet-sdk\artifacts\tasks\obj\toolset-tasks\Debug\net8.0\toolset-tasks.dll" to "E:\Code\dotnet-sdk\artifacts\tasks\bin\toolset-tasks\Debug\net8.0\toolset-tasks.dll". Beginning retry 4 in 1000ms. The process cannot access the file 'E:\Code\dotnet-sdk\artifacts\tasks\bin\toolset-tasks\Debug\net8.0\toolset-tasks.dll' because it is being used by another process. The file is locked by: ".NET Host (19284)" [E:\Code\dotnet-sdk\src\Layout\toolset-tasks\toolset-tasks.csproj]
    C:\Program Files\dotnet\sdk\9.0.100-alpha.1.24067.4\Microsoft.Common.CurrentVersion.targets(4787,5): warning MSB3026: Could not copy "E:\Code\dotnet-sdk\artifacts\tasks\obj\toolset-tasks\Debug\net8.0\toolset-tasks.dll" to "E:\Code\dotnet-sdk\artifacts\tasks\bin\toolset-tasks\Debug\net8.0\toolset-tasks.dll". Beginning retry 5 in 1000ms. The process cannot access the file 'E:\Code\dotnet-sdk\artifacts\tasks\bin\toolset-tasks\Debug\net8.0\toolset-tasks.dll' because it is being used by another process. The file is locked by: ".NET Host (19284)" [E:\Code\dotnet-sdk\src\Layout\toolset-tasks\toolset-tasks.csproj]
    C:\Program Files\dotnet\sdk\9.0.100-alpha.1.24067.4\Microsoft.Common.CurrentVersion.targets(4787,5): warning MSB3026: Could not copy "E:\Code\dotnet-sdk\artifacts\tasks\obj\toolset-tasks\Debug\net8.0\toolset-tasks.dll" to "E:\Code\dotnet-sdk\artifacts\tasks\bin\toolset-tasks\Debug\net8.0\toolset-tasks.dll". Beginning retry 6 in 1000ms. The process cannot access the file 'E:\Code\dotnet-sdk\artifacts\tasks\bin\toolset-tasks\Debug\net8.0\toolset-tasks.dll' because it is being used by another process. The file is locked by: ".NET Host (19284)" [E:\Code\dotnet-sdk\src\Layout\toolset-tasks\toolset-tasks.csproj]
    C:\Program Files\dotnet\sdk\9.0.100-alpha.1.24067.4\Microsoft.Common.CurrentVersion.targets(4787,5): warning MSB3026: Could not copy "E:\Code\dotnet-sdk\artifacts\tasks\obj\toolset-tasks\Debug\net8.0\toolset-tasks.dll" to "E:\Code\dotnet-sdk\artifacts\tasks\bin\toolset-tasks\Debug\net8.0\toolset-tasks.dll". Beginning retry 7 in 1000ms. The process cannot access the file 'E:\Code\dotnet-sdk\artifacts\tasks\bin\toolset-tasks\Debug\net8.0\toolset-tasks.dll' because it is being used by another process. The file is locked by: ".NET Host (19284)" [E:\Code\dotnet-sdk\src\Layout\toolset-tasks\toolset-tasks.csproj]
    C:\Program Files\dotnet\sdk\9.0.100-alpha.1.24067.4\Microsoft.Common.CurrentVersion.targets(4787,5): warning MSB3026: Could not copy "E:\Code\dotnet-sdk\artifacts\tasks\obj\toolset-tasks\Debug\net8.0\toolset-tasks.dll" to "E:\Code\dotnet-sdk\artifacts\tasks\bin\toolset-tasks\Debug\net8.0\toolset-tasks.dll". Beginning retry 8 in 1000ms. The process cannot access the file 'E:\Code\dotnet-sdk\artifacts\tasks\bin\toolset-tasks\Debug\net8.0\toolset-tasks.dll' because it is being used by another process. The file is locked by: ".NET Host (19284)" [E:\Code\dotnet-sdk\src\Layout\toolset-tasks\toolset-tasks.csproj]
    C:\Program Files\dotnet\sdk\9.0.100-alpha.1.24067.4\Microsoft.Common.CurrentVersion.targets(4787,5): warning MSB3026: Could not copy "E:\Code\dotnet-sdk\artifacts\tasks\obj\toolset-tasks\Debug\net8.0\toolset-tasks.dll" to "E:\Code\dotnet-sdk\artifacts\tasks\bin\toolset-tasks\Debug\net8.0\toolset-tasks.dll". Beginning retry 9 in 1000ms. The process cannot access the file 'E:\Code\dotnet-sdk\artifacts\tasks\bin\toolset-tasks\Debug\net8.0\toolset-tasks.dll' because it is being used by another process. The file is locked by: ".NET Host (19284)" [E:\Code\dotnet-sdk\src\Layout\toolset-tasks\toolset-tasks.csproj]
    C:\Program Files\dotnet\sdk\9.0.100-alpha.1.24067.4\Microsoft.Common.CurrentVersion.targets(4787,5): warning MSB3026: Could not copy "E:\Code\dotnet-sdk\artifacts\tasks\obj\toolset-tasks\Debug\net8.0\toolset-tasks.dll" to "E:\Code\dotnet-sdk\artifacts\tasks\bin\toolset-tasks\Debug\net8.0\toolset-tasks.dll". Beginning retry 10 in 1000ms. The process cannot access the file 'E:\Code\dotnet-sdk\artifacts\tasks\bin\toolset-tasks\Debug\net8.0\toolset-tasks.dll' because it is being used by another process. The file is locked by: ".NET Host (19284)" [E:\Code\dotnet-sdk\src\Layout\toolset-tasks\toolset-tasks.csproj]
    C:\Program Files\dotnet\sdk\9.0.100-alpha.1.24067.4\Microsoft.Common.CurrentVersion.targets(4787,5): error MSB3027: Could not copy "E:\Code\dotnet-sdk\artifacts\tasks\obj\toolset-tasks\Debug\net8.0\toolset-tasks.dll" to "E:\Code\dotnet-sdk\artifacts\tasks\bin\toolset-tasks\Debug\net8.0\toolset-tasks.dll". Exceeded retry count of 10. Failed. The file is locked by: ".NET Host (19284)" [E:\Code\dotnet-sdk\src\Layout\toolset-tasks\toolset-tasks.csproj]
    C:\Program Files\dotnet\sdk\9.0.100-alpha.1.24067.4\Microsoft.Common.CurrentVersion.targets(4787,5): error MSB3021: Unable to copy file "E:\Code\dotnet-sdk\artifacts\tasks\obj\toolset-tasks\Debug\net8.0\toolset-tasks.dll" to "E:\Code\dotnet-sdk\artifacts\tasks\bin\toolset-tasks\Debug\net8.0\toolset-tasks.dll". The process cannot access the file 'E:\Code\dotnet-sdk\artifacts\tasks\bin\toolset-tasks\Debug\net8.0\toolset-tasks.dll' because it is being used by another process. [E:\Code\dotnet-sdk\src\Layout\toolset-tasks\toolset-tasks.csproj]

Build failed with errors in 15.8s
```

What's especially gross here is that each copy error is repeated - in the TL display that's double bad because we never saw the copies in the first place.

### Analysis

_No response_

### Versions & Configurations

_No response_

### PR DESCRIPTION
**Pull request by surayya-MS**
Fixes #9658

### Old behavior

TerminalLoggerCopyRetry_OldBehavior

### New behavior

TerminalLoggerCopyRetry_NewBehavior

### CODE REVIEW

File: src/MSBuild/TerminalLogger/TerminalLogger.cs [hunk 1]
```diff
@@ -939,6 +940,9 @@ private void WarningRaised(object sender, BuildWarningEventArgs e)
         _immediateMessageKeywords.Any(imk => message.IndexOf(imk, StringComparison.OrdinalIgnoreCase) >= 0);
 #endif
 
+
+    private bool IsCopyTaskRetryCode(string code) => code == "MSB3026";
```
Comment (rainersigwald) [inline src/MSBuild/TerminalLogger/TerminalLogger.cs:944]: I suggest having a more generic name here, maybe something like

```suggestion
    private bool IsImmediateWarning(string code) =>
        // Copy task retries can often be resolved by closing an app, so surface them immediately.
        code == "MSB3026";
```

**Review by rainersigwald — APPROVED** [review]
Works for me! @baronfel this is going with the "special-case this code and punt the logging-API question" approach, which feels right to me at the moment.

File: src/MSBuild/TerminalLogger/TerminalLogger.cs (hunk 1 above)
Comment (surayya-MS) [reply src/MSBuild/TerminalLogger/TerminalLogger.cs]: done

**Comment by baronfel** [comment]
This is fine to fix the immediate problem, but I would like us to log an item to push for a 'full' fix - these messages looks noisy with the shared file prefix, and could overall do with a bit of cleanup presentation-wise when rendering in Terminal Logger. They look dated compared to how we render the project-specific errors.

**Review by MichalPavlik — APPROVED** [review]

**Pull request merged**